### PR TITLE
feat(debug): add debug:suppress_errors to suppress errors.

### PR DESCRIPTION
**Pull request by memchr**
If `debug:suppress_errors` is set to true, then `parseErrors` will not be sent to `CHyprError`.

Available in debug build only, no-op in release build

### CODE REVIEW

File: src/config/ConfigManager.cpp [hunk 1]
```diff
@@ -1566,7 +1567,11 @@ void CConfigManager::loadConfigLoadVars() {
         g_pHyprOpenGL->m_bReloadScreenShader = true;
 
     // parseError will be displayed next frame
+#ifdef HYPRLAND_DEBUG
+    if (parseError != "" && !configValues["debug:suppress_errors"].intValue)
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:1571]: no reason to guard this for only debug. `debug:` options are generally available.

Comment (memchr) [reply src/config/ConfigManager.cpp:1571]: My reasoning is that if some users used this option in the hyprland release build and forgot about it. then reported a issue when they found their configuration did not work...

Well, that would be very time-consuming.

Comment (memchr) [reply src/config/ConfigManager.cpp:1571]: So, should I remove the preprocessor guard?

Comment (vaxerski) [reply src/config/ConfigManager.cpp:1571]: yes

Comment (memchr) [reply src/config/ConfigManager.cpp:1570]: Removed the preprocessor guard, thus marked as resolved.

**Review by memchr — COMMENTED** [review]
As with the requirement, the preprocessor guard has been removed.

However, with the utmost respect, may I humbly suggest that consideration be given to adding a warning within the issue template, **advising users to check for and disable `debug:suppress_errors`** before submitting an issue report?

**Review by vaxerski — APPROVED** [review]
g

**Comment by vaxerski** [comment]
> However, with the utmost respect, may I humbly suggest that consideration be given to adding a warning within the issue template, **advising users to check for and disable `debug:suppress_errors`** before submitting an issue report?

We can. Although strict mentions in the wiki should suffice imo

**Pull request merged**